app/helpers/toolbar_helper.rb

```diff
@@ -44,9 +44,12 @@ def defered_toolbar_render(div_id, toolbar_name)
   #
   def render_toolbars
     @toolbars.collect do |div_id, toolbar_name|
-      content_tag(:div, :id => div_id) do # div for each toolbar
-        buttons = toolbar_name ? build_toolbar(toolbar_name) : nil
-        buttons_to_html(buttons)
+      buttons = toolbar_name ? build_toolbar(toolbar_name) : nil
+      visible_div = buttons.any? {|tb| tb[:hidden] == false} if buttons
```

Comment:
On top of that, you would need to hide the whole button group. See `split_to_groups` and `div.form-group`. Otherwise you would get empty toolbar sections between separators.